[docs] Migrate more pages (#243)
mui-org/material-ui-x: docs/src/pages/components/data-grid/overview/overview.md

The data grid comes in 2 versions:

> - `DataGrid` an **MIT license** as part of the community edition. It's an extension of `@material-ui/core`.

Reviewer: ```suggestion
- `DataGrid` **MIT licensed** as part of the community edition. It's an extension of `@material-ui/core`.
```

> - `XGrid` a **commercial license** as part of the X product line offering.

Reviewer: ```suggestion
- `XGrid` **Commercially licensed** as part of the X product line offering.
```

  The features only available in the commercial version are suffixed with a ⚡️ icon.

### MIT version

> The first version is meant to simplify the [Table demo](https://material-ui.com/components/tables/#sorting-amp-selecting) with a clean abtraction.

Reviewer: ```suggestion
The first version is meant to simplify the [Table demo](https://material-ui.com/components/tables/#sorting-amp-selecting) with a clean abstraction.
```

This doesn't seem like the right frame of reference.

## Features

- Built with and exclusively for React ⚛️
- High performance 🚀

> - Lightweight, less than [30 kB](https://bundlephobia.com/result?p=@material-ui/x-grid) gzipped with as few dependencies as possible

Reviewer: ```suggestion
- Lightweight; less than [30 kB](https://bundlephobia.com/result?p=@material-ui/x-grid) gzipped with as few dependencies as possible.
```
